pcsx2/Recording/InputRecordingControls.cpp

```diff
@@ -79,9 +79,12 @@ void InputRecordingControls::HandleFrameAdvanceAndPausing()
 		switchToReplay = false;
 	}
 
-	if ((g_InputRecording.IsReplaying() && g_InputRecording.GetFrameCounter() >= g_InputRecording.GetInputRecordingData().GetTotalFrames())
-		|| g_InputRecording.GetFrameCounter() == INT_MAX)
+	bool endCapture = false;
+	if (IsReplayingAtEnd() || g_InputRecording.GetFrameCounter() == INT_MAX)
+	{
 		pauseEmulation = true;
+		endCapture = true;
```

Comment:
It's been a minute since I touched this, but I believe it was a matter of making sure to wait until emulation is paused before calling for the capturing to stop. All other called to the function are in situations where emulation is already pre-paused.

Reply:
I wondered if that was the case, and yeah I later noticed the other functions setting this bool so probably nothing that needs to change. 👍